src/repositories/UniswapV3PoolRepository.ts

```diff
@@ -59,4 +64,60 @@ export class UniswapV3PoolRepository {
 
     return getModelForClass(UniswapV3Pool).find(filter).sort({token0: 1, token1: 1}).exec();
   }
+
+  async findBestPool(props: {
+    protocol: string;
+    fromChain: string[];
+    base: string;
+    quote: string;
+  }): Promise<UniswapV3Pool | undefined> {
+    const {protocol, fromChain, base, quote} = props;
+    const liquidityFreshness = this.getLiquidityFreshness(fromChain[0] as ChainsIds, protocol as DexProtocolName);
+    const liquidityUpdatedLimit = new Date(Date.now() - liquidityFreshness);
+
+    const filterToken0 = {
+      protocol,
+      chainId: {$in: fromChain},
+      token0: quote,
+      token1: base,
+      liquidityUpdatedAt: {$gt: liquidityUpdatedLimit},
+    };
+
+    const filterToken1 = {
+      protocol,
+      chainId: {$in: fromChain},
+      token0: base,
+      token1: quote,
+      liquidityUpdatedAt: {$gt: liquidityUpdatedLimit},
+    };
+
+    const [liquidityToken0, liquidityToken1] = await Promise.all([
+      getModelForClass(UniswapV3Pool).find(filterToken0).sort({liquidityLockedToken0: 1}).exec(),
+      getModelForClass(UniswapV3Pool).find(filterToken1).sort({liquidityLockedToken1: 1}).exec(),
+    ]);
+
+    if (liquidityToken0.length === 0 && liquidityToken1.length === 0) {
+      return undefined;
+    }
+
+    if (liquidityToken0.length > 0 && liquidityToken1.length === 0) {
+      return liquidityToken0[0];
+    }
+
+    if (liquidityToken1.length > 0 && liquidityToken0.length === 0) {
+      return liquidityToken1[0];
+    }
+
+    return liquidityToken0[0].liquidityLockedToken0 > liquidityToken1[0].liquidityLockedToken1
```

Comment:
this needs to be quote liquidity, you can add TODO and let's do it in next PR os we can merge current code

Reply:
Done